### PR TITLE
Add Profiler based code coverage

### PR DESCRIPTION
**Pull request by nohwnd**
## PR Summary

Add Profiler based code coverage and hide it under experimental UseBreakpoints flag. 

## PR Checklist

- [x] PR has meaningful title
- [x] Summary describes changes
- [ ] PR is ready to be merged
  - If not, use the arrow next to `Create Pull Request` to mark it as a draft. PR can be marked `Ready for review` when it's ready.
- [ ] Tests are added/update *(if required)*
- [x] Documentation is updated/added *(if required)*

### CODE REVIEW

**Comment by fflaten** [comment]
Just curious, but why the change in Be.ps1? You mentioned something about the host buffer width being affected by the custom PSHost somewhere, just can remember where and how you detected it. Can't reproduce it with e.g. `$host.UI.RawUI.BufferSize.Width; Trace-Script { Write-Warning "Inside Profiler - Width: $($host.UI.RawUI.BufferSize.Width)" }`

**Comment by nohwnd** [comment]
That change in Be is incorrect, I wanted to get whole output when comparing coverage files or something like that. Good catch.

File: src/functions/Coverage.ps1 [hunk 1]
```diff
@@ -1007,3 +1053,42 @@ function Add-JaCoCoCounter {
...
+            $assignment = $ast.Find( { param ($item) $item -is [System.Management.Automation.Language.AssignmentStatementAst] }, $false)
+            if ($assignment) {
+                if ($assignment.Right) {
+                    $hitColumn = $location.Column - $assignment.Right.Extent.StartColumnNumber + 1
+                }
+            }
+        }
+
+
+        $points.Add([Pester.Tracing.CodeCoveragePoint]::new($location.Script, $location.Line, $hitColumn, $location.Column, $breakpoint.Command));
```
Comment (nohwnd) [inline src/functions/Coverage.ps1:1081]: Yeah will have to fix those two `::new` for ps3.

**Comment by nohwnd** [comment]
More to fix:

- PS4 should use the PS5.1 implementation, not the ps3 implementation. 
```
[-] Context Code Coverage Analysis.Entire file measured using <description> failed
 InvalidCastException: Unable to cast object of type 'CallStackList' to type 'System.Collections.IList'.
 MethodInvocationException: Exception calling "Patch" with "4" argument(s): "Unable to cast object of type 'CallStackList' to type 'System.Collections.IList'."
```

- PS6_2:
```
[-] Context Code Coverage Analysis.Entire file measured using <description> failed
 NullReferenceException: Object reference not set to an instance of an object.
 MethodInvocationException: Exception calling "Patch" with "4" argument(s): "Object reference not set to an instance of an object."
```
Either some of the fields was renamed, or I am just using the wrong name in the if powershell 7 field switch.

**Pull request merged**